Update the Go version to 1.22

robertsirc opened the pull request:
This change is to update the go version from 1.21 to 1.22 and the golangci-lint from 1.55 to 1.56.

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/main/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

robertsirc committed:
Update the Go version to 1.22

This change is to update the go version from 1.21 to 1.22 and the golangci-lint from 1.55 to 1.56.

Signed-off-by: Robert Sirchia <rsirchia@outlook.com>

mattfarina commented on .github/workflows/golangci-lint.yml:
```diff
@@ -19,4 +19,4 @@ jobs:
         uses: golangci/golangci-lint-action@3a919529898de77ec3da873e3063ca4b10e7f5cc #pin@3.7.0
         with:
-          version: v1.55
+          version: v1.56
```
Earlier in this file there is a line that reads

```
go-version: "1.21"
```

This needs to be updated to 1.22.

robertsirc replied:
Silly mistake sorry.

mattfarina commented on go.mod:
```diff
@@ -1,6 +1,6 @@
 module helm.sh/helm/v3
 
-go 1.21
+go 1.22
```
It appears the [codeql](https://codeql.github.com/) (security scanner) does not support Go 1.22 being declared in the `go.mod` file. The change has been made but not yet released.

Can you undo this change for now? We can make it later.

robertsirc replied:
Reverting this now.

mattfarina reviewed (commented):
I think I found why the tests are failing...

robertsirc committed:
Fixing issues as per feedback

Moved go.mod back to 1.21 due to code ql and actually changing the version of go in golangci-lint.

Signed-off-by: Robert Sirchia <rsirchia@outlook.com>

robertsirc committed:
Merge branch 'helm:main' into Update-Go-to-1.22-for-the-builds

robertsirc commented:
@mattfarina should I abandon this after our discussion?
